binary-oracle-pair/program/src/instruction.rs

```diff
@@ -0,0 +1,96 @@
+//! Instruction types
+
+//use crate::{
+//    error::PoolError,
+//};
+//use num_derive::{FromPrimitive, ToPrimitive};
+//use num_traits::{FromPrimitive, ToPrimitive};
+use solana_program::{
+    clock::Slot,
+    //sysvar,
+    //instruction::{AccountMeta},
+    program_error::ProgramError,
+    program_pack::{IsInitialized, Pack, Sealed},
+};
+
+//use std::{convert::TryInto, mem::size_of};
+
+/// Instruction definition
+#[derive(Clone, Debug, PartialEq)]
+pub enum Instruction {
+    /// Initializes a new binary oracle pair pool.
+    ///
+    ///   0. `[writable]` Pool account.
+    ///   1. `[]` authority create_program_address(&[Token-swap account])`
+    ///   2. `[]` Decider authority
```

Comment:
Also, we'll need to build in the ability to halt trading on the swap -- basic admin functionality

Reply:
this doesn't need to have the same authority as the AMM/swap market.  once the tokens are minted the user can convert them on the market to the position they want, and arbs/bots can mint more to balance it out.

Reply:
You can strike this one too, I thought a token swap was inherently tied to this